Tests/SPTPersistentCachePerformance.m

```diff
@@ -0,0 +1,213 @@
+#import <XCTest/XCTest.h>
+#import <mach/mach_time.h>
+
+#import <SPTPersistentCache/SPTPersistentCache.h>
+double convertMachToMilliSeconds(uint64_t mach_time);
+
+static const char* kFileNames[] = {
+    "aad0e75ab0a6828d0a9b37a68198cc9d70d84850",
+    "ab3d97d4d7b3df5417490aa726c5a49b9ee98038",
+    "b02c1be08c00bac5f4f1a62c6f353a24487bb024",
+    "b3e04bf446b486412a13659af71e3a333c6152f4",
+    "b53aed36cdc67dd43b496db74843ac32fe1f64bb",
+    "b91998ae68b9639cee6243df0886d69bdeb75854",
+    "c3b19963fc076930dd36ce3968757704bbc97357",
+    "c5aec3eef2478bfe47aef16787a6b4df31eb45f2",
+    "e5a1921f8f75d42412e08aff4da33e1132f7ee8a",
+    "e5b8abdc091921d49e86687e28b74abb3139df70",
+    "ee6b44ab07fa3937a6d37f449355b64c09677295",
+    "ee678d23b8dba2997c52741e88fa7a1fdeaf2863",
+    "eee9747e967c4440eb90bb812d148aa3d0056700",
+    "f1eeb834607dcc2b01909bd740d4356f2abb4cd1",
+    "f7501f27f70162a9a7da196c5d2ece3151a2d80a",
+    "f50512901688b79a7852999d384d097a71fad788",
+    "fc22d4f65c1ba875f6bb5ba7d35a7fd12851ed5c",
+    NULL
+};
+
+static const int SPTPersistentCachePerformanceIterationCount = 200;
+
+@interface SPTPersistentCacheTiming : NSObject
+@property (nonatomic) uint64_t queueTime;
+@property (nonatomic) uint64_t startTime;
+@property (nonatomic) uint64_t endTime;
+
+- (double) calculateTimeInQueue;
+- (double) calculateTimeExecuting;
+- (double) calculateTotalTime;
+@end
+
+@implementation SPTPersistentCacheTiming
+
+- (double) calculateTimeInQueue
+{
+    return convertMachToMilliSeconds(self.startTime) - convertMachToMilliSeconds(self.queueTime);
+}
+- (double) calculateTimeExecuting
+{
+    return convertMachToMilliSeconds(self.endTime) - convertMachToMilliSeconds(self.startTime);
+}
+- (double) calculateTotalTime
+{
+    return convertMachToMilliSeconds(self.endTime) - convertMachToMilliSeconds(self.queueTime);
+}
+
+@end
+
+@interface SPTPersistentCachePerformance : XCTestCase
+
+
+@property (nonatomic, strong) SPTPersistentCache *dataCache;
+@property (nonatomic, strong) NSMutableArray<NSData *> *fileContents;
+@property (nonatomic, strong) NSMutableArray<SPTPersistentCacheTiming *> *unLockTimings;
+@property (nonatomic, strong) NSMutableArray<SPTPersistentCacheTiming *> *readTimings;
+@property (nonatomic, strong) NSString *cachePath;
+
+@end
+
+@implementation SPTPersistentCachePerformance
+
+- (void)setUp
+{
+    [super setUp];
+    self.fileContents = [NSMutableArray array];
+
+    self.cachePath = [NSTemporaryDirectory() stringByAppendingPathComponent:
+                      [NSString stringWithFormat:@"pdc-%@.tmp", [[NSProcessInfo processInfo] globallyUniqueString]]];
+
+    NSLog(@"%@", self.cachePath);
+
+    NSBundle *bundle = [NSBundle bundleForClass:[self class]];
+    int i = 0;
+    while (kFileNames[i] != NULL) {
```

Comment:
Instead of putting NULL in the array you can use the sizeof() call since it is statically allocated